### ISSUE TITLE
generate image timeout

### ISSUE THREAD

**Issue by zc9** [frinyvonnick/node-html-to-image#179]
"node-html-to-image": "^3.2.4"
node: 16.5.0
os: mac os 12.3.1

I am trying to generate images with loop:

for (let i =0; i<10; i++){
 await nodeHtmlToImage({
      puppeteerArgs: {
        timeout: 0,
        args: ['--no-sandbox', '--disable-setuid-sandbox'],
        headless: false,
      },
      output: outpoutPath,
      html: html_static_data,
      content: {
           ...
      },
    });
}
But almost showing up the same error:

Error: Timeout hit: 30000
    at  xxxxx/node_modules/puppeteer-cluster/src/util.ts:73:19
    at Generator.next (<anonymous>)
    at fulfilled (/xxxxx/node_modules/puppeteer-cluster/dist/util.js:5:58)
So How I can set up the timeout to resolve this error?

Thanks!

**Comment by silhouette5366** [comment]
having the same issue, did anyone find any solution?

**Comment by adasq** [comment]
any updates? 🤔

**Comment by frinyvonnick** [comment]
node-html-to-image uses puppeteer. You can pass any options you want with the property puppeteerArgs. Here is the documentation about timeout option https://pptr.dev/api/puppeteer.launchoptions.timeout/

**Comment by zjacky0409** [comment]
> node-html-to-image uses puppeteer. You can pass any options you want with the property puppeteerArgs. Here is the documentation about timeout option https://pptr.dev/api/puppeteer.launchoptions.timeout/

Thank you for your help.

**Comment by adasq** [comment]
@frinyvonnick have you tried to pass this parameter with a success? It doesn't work for me.

**Comment by frinyvonnick** [comment]
@adasq could you provide the exact steps and code to reproduce your issue please?

**Comment by adasq** [comment]
Essentially, here's a snippet that forces timeout:

```    
    // create HTTP server
    import Hapi from '@hapi/hapi';

    const server = Hapi.server({
        port: process.env.PORT || 3000,
    });

    // The image (http://localhost:3000/img.png) loads after 40 seconds

    server.route({
        method: 'GET',
        path: '/img.png',
        handler: async (request, h) => {
            h.response('image/png');
            return new Promise((resolve) => setTimeout(() => {
                resolve()
            }, 40 * 1000))
        }
    });
    await server.start();

    // try to render
    return await nodeHtmlToImage({
        quality: 100,
        type: 'png',
        transparent: true,
        waitUntil: 'load',
        puppeteerArgs: {
            headless: true,
            timeout: 36000,
            args: [
                '--no-sandbox',
                '--disable-setuid-sandbox',
            ]
        },
        content: {},
        // try to render a page with a img
        html: '<html> <body><img src="http://localhost:3000/img.png" /> </body> </html>'
    })
```

Even though `timeout` is passed (`36000`), I get a default timeout `30000`:
![image](https://user-images.githubusercontent.com/5637734/213180708-121aab42-05be-481f-b824-fb91c584dd86.png)

**Comment by adasq** [comment]
I've noticed that you use `puppeteer-cluster`:
https://github.com/frinyvonnick/node-html-to-image/blob/master/src/main.ts#L21

The library defines its own timeout as well:
https://github.com/thomasdondorf/puppeteer-cluster#clusterlaunchoptions (see `timeout` option) that defauls to `30` sec.

The option isn't overridden by your package, though. I tried to specify it on my own by inlining in, but it didn't help either. 🤔

**Comment by frinyvonnick** [comment]
> I've noticed that you use `puppeteer-cluster`: https://github.com/frinyvonnick/node-html-to-image/blob/master/src/main.ts#L21
> 
> The library defines its own timeout as well: https://github.com/thomasdondorf/puppeteer-cluster#clusterlaunchoptions (see `timeout` option) that defauls to `30` sec.
> 
> The option isn't overridden by your package, though. I tried to specify it on my own by inlining in, but it didn't help either. 🤔

You were quicker to post your answer than me! In fact it seems to override the one from puppeteer 😢 I'll investigate when I have free time but it might be in a long time. If you find a solution in the meantime feel free to open a pull request 👍

**Comment by ariv803** [comment]
@frinyvonnick So this issue is still present? I dont understand the last couple comments. Can we specify the timeout or is it defined by puppeteer-cluster and we cant change it currently??

**Comment by ariv803** [comment]
Looks like this issue is still present and the option to use a timeout was added in a separate npm module called nhti as described here: https://github.com/frinyvonnick/node-html-to-image/pull/189  . The issue we have is when the timeout is hit it completely crashes the server and express automatically restarts itself. Any guidance would be great. Thank you!

**Comment by frinyvonnick** [comment]
The pull request #189 has been merged and released in [4.0.0](https://github.com/frinyvonnick/node-html-to-image/releases/tag/v4.0.0) 🚀
